internal/tracker/source/tekton/tekton.go

```diff
@@ -348,23 +352,50 @@ func PreparePackage(i *PreparePackageInput) (*hub.Package, error) {
 	var name, version, description, tektonKind string
 	var annotations map[string]string
 	var tasks []map[string]interface{}
+	var containerImages []*hub.ContainerImage
+
 	switch m := i.Manifest.(type) {
 	case *v1beta1.Task:
 		tektonKind = "task"
 		name = m.Name
 		version = m.Labels[versionLabelTKey]
 		description = m.Spec.Description
 		annotations = m.Annotations
+
+		// Prepare container images
+		ts := m.TaskSpec()
+		var defaults []v1beta1.ParamSpec
+		if len(ts.Params) > 0 {
+			defaults = append(defaults, ts.Params...)
+		}
+		mts := task.ApplyParameters(context.Background(), &ts, &v1beta1.TaskRun{}, defaults...)
+		for _, s := range mts.Steps {
+			if _, err := n.ParseReference(s.Image); s.Image != "" && err == nil {
```

Comment:
Thanks for the feedback! After giving it a bit more thoughts, I think we should still process and surface the catalog even if some container images may not be valid (and the valid container images should still be scanned). Pipeline/Task validation should be done on Tekton side and we will provide testing infra.

In this case I think it makes more sense to drop the `pkg.ValidateContainersImages` and keep this part as it is?

Reply:
No problem. Please note that the catalog would still be processed, it's just the task/pipeline with the invalid image the one that wouldn't, reporting the error to the publisher. Otherwise yes, we may not need the call to `pkg.ValidateContainersImages` for now.

Reply:
SG, I think we can leverage the reporting feature of Artifact Hub to start with and collect some user feedback. Still keeped `pkg.ValidateContainersImages` and just checked empty iamges beforehand